### ISSUE TITLE
Commit message intermittently missing package name

### ISSUE THREAD

**Issue by tsmalls93** [bluebrown/kobold#66]
I have noticed on a couple of occasions that the commit message is just "chore(kobold): ". I am not sure of the cause. The issue appears to be intermittent, because often when I re-trigger the webhook it works.

**Comment by bluebrown** [comment]
Hi, thanks for reporting. There was a pull request recently regarding to the messages. It was a bit messy. I tried to clean up afterwards but it may have some bugs.

**Comment by bluebrown** [comment]
related code is around here https://github.com/bluebrown/kobold/blob/d77d8770d849273edb42fff63fcb3ddae81292ab/krm/filter.go#L51

and here https://github.com/bluebrown/kobold/blob/d77d8770d849273edb42fff63fcb3ddae81292ab/task/handler.go#L48

**Comment by rossigee** [comment]
Maybe related #67

**Comment by tsmalls93** [comment]
Apologies for the delay in response. I have confirmed that this bug is still occurring in 0.4.1.

**Comment by bluebrown** [comment]
Yeah I figured that missing error handle wasnt the cause. Its very unlikely it accually error something like strings.Builder.Write, or what it was. 

Does it actually change images when it commits those messages?

**Comment by tsmalls93** [comment]
Yeah, it still updates the images.

**Comment by tsmalls93** [comment]
Unfortunately, it looks like this is still an issue.

<img width="401" alt="Screenshot 2024-12-16 at 9 22 27 AM" src="https://github.com/user-attachments/assets/1c4fd71b-5504-416c-ad1d-ae0ec59ae437" />

**Comment by karnirathorewolf** [comment]
In the `Filter()` method of `ImageRefUpdateFilter`, the code iterates through multiple image references and invokes the handler for each one:

[krm/filter.go](krm/filter.go#L132-L134)

```go
for _, imageRef := range i.imageRefs {
    v, change, err := i.handler(mn.Key.YNode().Value, mn.Value.YNode().Value, imageRef, opts)
    if err != nil {
        // ... error handling
        continue
    }
    mn.Value.YNode().Value = v
    lastChange = change  // BUG: Overwrites even if change is empty
}
```

**The bug:** When a handler returns a non-matching result, it returns an empty `Change{}`. Without a guard, this empty change would unconditionally overwrite the `lastChange` variable that contained the metadata from a previous successful match.

**Scenario:**
1. Process imageRef A (nginx:latest) → matches → `lastChange = {Description: "update image ref...", Registry: "...", Repo: "..."}`
2. Process imageRef B (busybox:latest) → doesn't match → returns `Change{}`
3. Without guard: `lastChange` becomes empty, losing all metadata
4. Later: only the empty `Change{}` is recorded → no change information

## Solution

Added a guard condition to only update `lastChange` when the handler returns a meaningful change:

```go
if change.Description != "" {
    lastChange = change
}
```

Related [PR 77](https://github.com/bluebrown/kobold/pull/77)
